src/Counters.cpp

```diff
@@ -26,15 +26,17 @@ Counters::Counters() : m_counters(STATS_END)
 {
 }
 
-unsigned& Counters::operator[](size_t index)
+unsigned&
+Counters::operator[](size_t index)
```

Comment:
If you rebase to latest master you'll see that this fails check_format. Or is it only here to trigger the failure?

Reply:
This was auto formatted by clang-format-10.
I'll try rebasing to see what happens.

Reply:
Rebased on current master (606bf51) and force-pushed (498a2d5):

    alex@DESKTOP-ALEX:~/ccache$ make check_format
    alex@DESKTOP-ALEX:~/ccache$ make format
    alex@DESKTOP-ALEX:~/ccache$ git status
    On branch clang-tidy
    nothing to commit, working tree clean
    alex@DESKTOP-ALEX:~/ccache$ clang-format --version
    clang-format version 10.0.0

Reply:
How annoying. Looks like they fixed a bug in clang-format 10, but we can't really require clang-format 10. This makes the CI check kind of useless unless we start requiring clang-format 6-9 or start bundling (or making it easy to download) a clang-format 10 binary.

I guess the same thing goes for clang-tidy in principle.

Reply:
Changing AlwaysBreakAfterReturnType from AllDefinitions to All just made the declarations **fail**:
[Code](https://github.com/ccache/ccache/compare/41b206f...AlexanderLanin:test), [Travis](https://travis-ci.com/github/AlexanderLanin/ccache/jobs/306889601)

Changing AlwaysBreakAfterReturnType from AllDefinitions to None makes it **pass**:
 [Code](https://github.com/AlexanderLanin/ccache/compare/41b206f...AlexanderLanin:test3), [Travis](https://travis-ci.com/github/AlexanderLanin/ccache/builds/156151361)

Adding old option AlwaysBreakAfterDefinitionReturnType made no difference, **fail**:
[Code](https://github.com/ccache/ccache/compare/41b206f...AlexanderLanin:test2), [Travis](https://travis-ci.com/github/AlexanderLanin/ccache/jobs/306889887)

Adding comments to those two places made it **fail**:
[Code](https://github.com/ccache/ccache/compare/41b206f...8acd1e9), [Travis](https://travis-ci.com/github/AlexanderLanin/ccache/jobs/306891547)

Moved comments by one line made it **pass**:
[Code](https://github.com/ccache/ccache/compare/41b206f...270187f), [Travis](https://travis-ci.com/github/AlexanderLanin/ccache/builds/156152340)

Reply:
Bundling clang-format for every system is probably a huge job.
Maybe format/check_format scripts should just fail when clang-format version x is not available since ccache (travis) is using version x? Maybe it's not 10 but an older version like 7 or 8 which had the bugfix. Such an old version could be more realistically enforced?

For now one of the two working workarounds on top seem sufficient.

Reply:
> Maybe it's not 10 but an older version like 7 or 8 which had the bugfix.

No, versions 6-9 are stable in this regard. I've access to them all (but not 10).

> For now one of the two working workarounds on top seem sufficient.

OK, `// clang-format off/on` comments will have to be good enough. I already had to use them in `CacheEntryWriter.cpp` to work around a clang-format bug.